shell/platform/android/io/flutter/embedding/android/FlutterFragment.java

```diff
@@ -76,219 +72,256 @@
+   * @param flutterShellArgs any special configuration arguments for the Flutter engine
+   *
+   * @return a new {@link FlutterFragment}
+   */
   public static FlutterFragment newInstance(boolean isSplashScreenDesired,
                                             @Nullable String initialRoute,
-                                            @NonNull String appBundlePath) {
+                                            @Nullable String appBundlePath,
+                                            @Nullable String dartEntrypoint,
+                                            @Nullable FlutterShellArgs flutterShellArgs) {
     FlutterFragment frag = new FlutterFragment();
 
+    Bundle args = createArgsBundle(
+        isSplashScreenDesired,
+        initialRoute,
+        appBundlePath,
+        dartEntrypoint,
+        flutterShellArgs
+    );
+    frag.setArguments(args);
+
+    return frag;
+  }
+
+  /**
+   * Creates a {@link Bundle} or arguments that can be used to configure a {@link FlutterFragment}.
+   * This method is exposed so that developers can create subclasses of {@link FlutterFragment}.
+   * Subclasses should declare static factories that use this method to create arguments that will
+   * be understood by the base class, and then the subclass can add any additional arguments it
+   * wants to this {@link Bundle}. Example:
+   *
+   * <pre>{@code
+   * public static MyFlutterFragment newInstance(String myNewArg) {
+   *   // Create an instance of our subclass Fragment.
+   *   MyFlutterFragment myFrag = new MyFlutterFragment();
+   *
+   *   // Create the Bundle or args that FlutterFragment understands.
+   *   Bundle args = FlutterFragment.createArgsBundle(...);
+   *
+   *   // Add our new args to the bundle.
+   *   args.putString(ARG_MY_NEW_ARG, myNewArg);
+   *
+   *   // Give the args to our subclass Fragment.
+   *   myFrag.setArguments(args);
+   *
+   *   // Return the newly created subclass Fragment.
+   *   return myFrag;
+   * }
+   * }</pre>
+   *
+   * @param isSplashScreenDesired should a splash screen be shown until the 1st Flutter frame is rendered?
+   * @param initialRoute the first route that a Flutter app will render in this {@link FlutterFragment}, defaults to "/"
+   * @param appBundlePath the path to the app bundle which contains the Dart app to execute
+   * @param dartEntrypoint the name of the initial Dart method to invoke, defaults to "main"
+   * @param flutterShellArgs any special configuration arguments for the Flutter engine
+   *
+   * @return Bundle of arguments that configure a {@link FlutterFragment}
+   */
+  public static Bundle createArgsBundle(boolean isSplashScreenDesired,
+                                        @Nullable String initialRoute,
+                                        @Nullable String appBundlePath,
+                                        @Nullable String dartEntrypoint,
+                                        @Nullable FlutterShellArgs flutterShellArgs) {
```

Comment:
Are there ever cases we expect this to be called directly by app developers, as opposed to this class or subclasses?

Reply:
Those cases are the same: app developers will call this when creating subclasses